### PR TITLE
port the router to solc0.8

### PR DESCRIPTION
**Pull request by bixia**
hi, i basically port the UniswapV2Router02.sol to the current repo. please help to review

### CODE REVIEW

**Comment by islishude** [comment]
Hi @bixia , Thanks for the PR.

Can you fix the ci issues?

File: contracts/libraries/SafeMath.sol [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.6.0;
```
Comment (islishude) [inline contracts/libraries/SafeMath.sol:3]: Solidity 0.8 has built-in SafeMath, there is no need to include SafeMath library

File: contracts/UniswapV2Factory.sol [hunk 2]
```diff
@@ -56,4 +56,8 @@ contract UniswapV2Factory is IUniswapV2Factory {
         require(msg.sender == feeToSetter, "UniswapV2: FORBIDDEN");
         feeToSetter = _feeToSetter;
     }
+    function getPairHash() external returns (bytes32) {
```
Comment (islishude) [inline contracts/UniswapV2Factory.sol:59]: I think it should be pure function.

It changes UniswapV2Fatory, Could you add an uint test for this?

**Comment by RauppRafael** [comment]
Hi, I can help with this PR, would be nice to first merge the original periphery repo so we can see exactly what has changed, what do you think @islishude ?

File: contracts/UniswapV2Factory.sol (hunk 2 above)
Comment (RauppRafael) [reply contracts/UniswapV2Factory.sol:59]: It's better to add this to the top of the file as a constant:

`bytes32 public constant PAIR_HASH = keccak256(abi.encodePacked(type(GenesisPair).creationCode));`

**Comment by islishude** [comment]
> Hi, I can help with this PR, would be nice to first merge the original periphery repo so we can see exactly what has changed, what do you think @islishude ?

Cool, I will check it out later

File: contracts/UniswapV2Factory.sol [hunk 3]
```diff
@@ -56,4 +56,8 @@ contract UniswapV2Factory is IUniswapV2Factory {
...
+        //0x3bdbb0f75665853c99209993e6c3464df4ca1c8a173e9f0b69447f56d617a676
```
[the remark this answers is not in the document]
Comment (bixia) [reply contracts/UniswapV2Factory.sol:60]: Yes, agree with that

**Comment by islishude** [comment]
Check out new PR #7

**Pull request closed without merging**